llm_studio/src/utils/config_utils.py

```diff
@@ -208,25 +200,12 @@ def load_config_yaml(path: str) -> ConfigProblemBase:
     with open(path, "r") as fp:
         cfg_dict = yaml.load(fp, Loader=yaml.FullLoader)
 
-    cfg = ConfigProblemBase(
-        output_directory=cfg_dict.get(
-            "output_directory", ConfigProblemBase.output_directory
-        ),
-        experiment_name=cfg_dict.get("experiment_name", generate_experiment_name()),
-        llm_backbone=cfg_dict.get("llm_backbone", ConfigProblemBase.llm_backbone),
-        dataset=ConfigNLPCausalLMDataset.from_dict(cfg_dict.get("dataset", {})),
-        tokenizer=ConfigNLPCausalLMTokenizer.from_dict(cfg_dict.get("tokenizer", {})),
-        augmentation=ConfigNLPAugmentation.from_dict(cfg_dict.get("augmentation", {})),
-        architecture=ConfigNLPCausalLMArchitecture.from_dict(
-            cfg_dict.get("architecture", {})
-        ),
-        training=ConfigNLPCausalLMTraining.from_dict(cfg_dict.get("training", {})),
-        prediction=ConfigNLPCausalLMPrediction.from_dict(
-            cfg_dict.get("prediction", {})
-        ),
-        environment=ConfigNLPCausalLMEnvironment.from_dict(
-            cfg_dict.get("environment", {})
-        ),
-        logging=ConfigNLPCausalLMLogging.from_dict(cfg_dict.get("logging", {})),
-    )
-    return cfg
+    if cfg_dict["problem_type"] == "text_causal_language_modeling":
```

Comment:
Shall we somewhere note that this needs to be added for new problem types? But probably it will be mostly us and we will hopefully remember or notice.

Reply:
Good point! One could think about rewriting this or add better documentation. I'd propose to leave it as is for now and potentially refactor once we add more problem types.